coffee-chats/src/main/webapp/src/entity/Requests.ts

```diff
@@ -0,0 +1,167 @@
+interface RequestParams {
+  tags: string[],
+  dateRanges: DateRange[],
+  minPeople: number,
+  maxPeople: number,
+  durationMins: number,
+  matchRandom: boolean,
+  matchRecents: boolean,
+  userId: string
+}
+
+/**
+ * Exposed interfaces for using and managing entites
+ */
+export interface ChatRequest extends RequestParams {
+  requestId: number
+};
+
+export interface MatchedRequest {
+  matched: boolean,
+  dateRange: DateRange,
+  durationMins: number,
+  userId: string,
+  tags: string[],
+  participants: string[],
+  commonTags: string[]
+};
+
+export interface ExpiredRequest extends RequestParams {
+  matched: boolean
+};
+
+export type CompletedRequest = ExpiredRequest | MatchedRequest;
+
+export interface DateRange {
+  start: Date,
+  end: Date
+};
+
+
+/**
+ * Internal interfaces for handling JSON format recieved via fetch requests.
+ */
+interface RequestParamsResponse {
+  tags: string[],
+  dateRanges: DateRangeResponse[],
+  minPeople: number,
+  maxPeople: number,
+  duration: DurationResponse,
+  matchRandom: boolean,
+  matchRecents: boolean,
+  userId: string
+}
+
+export interface ChatRequestResponse extends RequestParamsResponse{
```

Comment:
missing space

why you don't tell automation to format code for you?

Reply:
I previously had done so, however it was not working well with the react TSX elements and breaking the spacing within those. Perhaps with more time I would configure this properly.